changelog/6384.bugfix.rst

```diff
@@ -0,0 +1 @@
+Make `--showlocals` work also with `--tb=short`.
```

Comment:
* I think this should be an `improvement`, the code clearly intended to hide locals when using `--tb=short`. 

* Because of the above, this should target `features`.

Other than that LGTM 👍

Reply:
Moved to features.